restapi/src/main/java/io/stargate/web/resources/v2/schemas/UdtsResource.java

```diff
@@ -0,0 +1,380 @@
+      @ApiParam(value = "Name of the udt.", required = true) @PathParam("typeName")
+          final String typeName,
+      @ApiParam(
+              defaultValue = "false",
+              value =
+                  "Attempting to drop a non existing udt returns an error unless this option is true.")
+          @QueryParam("ifExists")
+          final boolean ifExists,
+      @Context HttpServletRequest request) {
+    return RequestHandler.handle(
+        () -> {
+          Map<String, String> allHeaders = getAllHeaders(request);
+          AuthenticatedDB authenticatedDB = db.getDataStoreForToken(token, allHeaders);
+
+          db.getAuthorizationService()
+              .authorizeSchemaWrite(
+                  authenticatedDB.getAuthenticationSubject(),
+                  keyspaceName,
+                  null,
+                  Scope.DROP,
+                  SourceAPI.REST,
+                  ResourceKind.TYPE);
+
+          Keyspace keyspace = authenticatedDB.getDataStore().schema().keyspace(keyspaceName);
+          if (keyspace == null) {
+            return Response.status(Response.Status.NOT_FOUND)
+                .entity(
+                    new Error(
+                        String.format("Keyspace '%s' not found.", keyspaceName),
+                        Response.Status.NOT_FOUND.getStatusCode()))
+                .build();
+          }
+
+          try {
+            authenticatedDB
+                .getDataStore()
+                .queryBuilder()
+                .drop()
+                .type(keyspaceName, UserDefinedType.reference(keyspaceName, typeName))
+                .ifExists(ifExists)
+                .build()
+                .execute(ConsistencyLevel.LOCAL_QUORUM)
+                .get();
+          } catch (IllegalArgumentException | NotFoundException iae) {
+            return Response.status(Response.Status.BAD_REQUEST)
+                .entity(
+                    new Error(
+                        String.format("%s", iae.getMessage()),
+                        Response.Status.BAD_REQUEST.getStatusCode()))
+                .build();
+          }
+          return Response.status(Response.Status.NO_CONTENT).build();
+        });
+  }
+
+  @Timed
+  @POST
+  @ApiOperation(
+      value = "Create an UDT",
+      notes = "Create an UDT in the given keyspace",
+      response = String.class,
+      responseContainer = "Map",
+      code = 201)
+  @ApiResponses(
+      value = {
+        @ApiResponse(
+            code = 201,
+            message = "resource created",
+            response = Map.class,
+            responseContainer = "Map"),
+        @ApiResponse(code = 400, message = "Bad Request", response = Error.class),
+        @ApiResponse(code = 401, message = "Unauthorized", response = Error.class),
+        @ApiResponse(code = 409, message = "Conflict", response = Error.class),
+        @ApiResponse(code = 500, message = "Internal server error", response = Error.class)
+      })
+  @Path("/{typeName}")
```

Comment:
Since this is a POST it would make more sense to have the name in the body. This is implying that a resource already exists at `/typeName`